vega_sim/service.py

```diff
@@ -1169,17 +1170,28 @@ def settle_market(
     def party_account(
         self,
         key_name: str,
-        asset_id: str,
-        market_id: str,
+        market_id: Optional[str] = None,
+        asset_id: Optional[str] = None,
         wallet_name: Optional[str] = None,
     ) -> data.PartyMarketAccount:
         """Output money in general accounts/margin accounts/bond accounts (if exists)
         of a party."""
-        return data.party_account(
-            self.wallet.public_key(wallet_name=wallet_name, name=key_name),
-            asset_id=asset_id,
+        if market_id is None and asset_id is None:
+            raise Exception("Either market or asset must be passed")
+        if market_id is not None and asset_id is not None:
+            raise Exception("Only one of market or asset can be passed")
```

Comment:
Think these might be causing some of the failures, what's the purpose of this check?

Reply:
Failures all fixed up now. There are a few different ways one could be wanting data from this and it's tricky to disambiguate them all! This was pretty much just removing one ambiguity (where you want everything, which I've added a function for in local data cache but haven't bothered piping through yet)